### PR TITLE
[HOTFIX] Fix InsertFromStage complex data type issue for partition table

### PR DESCRIPTION
**Pull request by niuge01**
### Why is this PR needed?
 CarbonInsertFromStageCommand with complex data type are not working fine.
 
 ### What changes were proposed in this PR?
For the partition table, the complex data type of target table should be converted to the binary data type.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by niuge01** [comment]
please test this

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1404/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1403/

**Comment by CarbonDataQA1** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1413/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1426/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonLoadDataCommand.scala [hunk 1]
```diff
@@ -872,13 +851,19 @@ case class CarbonLoadDataCommand(
     val table = loadModel.getCarbonDataLoadSchema.getCarbonTable
     val metastoreSchema = StructType(catalogTable.schema.fields.map{f =>
       val column = table.getColumnByName(f.name)
-      if (column.hasEncoding(Encoding.DICTIONARY)) {
-        f.copy(dataType = IntegerType)
-      } else if (f.dataType == TimestampType || f.dataType == DateType) {
-        f.copy(dataType = LongType)
+      val updatedDataType = if (column.hasEncoding(Encoding.DICTIONARY)) {
```
Comment (QiangCai) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonLoadDataCommand.scala]: please rebase code

Comment (niuge01) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonLoadDataCommand.scala]: done

**Comment by niuge01** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1410/

**Comment by CarbonDataQA1** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1418/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1412/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1432/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1434/

**Comment by ydvpankaj97** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1420/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**